### DISCUSSION TITLE
Connection limit on a node with multiple handles

### DISCUSSION THREAD

**Discussion by badiuciprian** [xyflow/xyflow#4122] [Q&A]
Hi, I just started to use this awesome tool and it's great, I just have an use-case that I cannot figure out how to implement. I have custom nodes with multiple source handles and target handles (eg. one node with 3 inputs and 2outputs) and I need to limit the connections for target handles to one (so each target should only accept one connection, but the sources can connect to any number of nodes). I found the example on how to limit the connection number, but it limits it to 1 per node, not per handle. Any ideas on how to solve this?

**Comment by bcakmakoglu** [comment]
Have you given the [`Connection Limit`](https://reactflow.dev/examples/nodes/connection-limit) example a look?

**Reply by badiuciprian** [reply]
That was a fast response :) yes, and it works fine as long as you have a single handle on the node. But I can have multiple handles on my custom node. For example 3targets and 2 sources, and I need to limit the connections to 1per target handle, not per node.  Following the example I can limit the connections to 1 per node, but then the other target handles get unconnectable

**Reply by bcakmakoglu** [reply]
The same concept applies though - check how many connections for a specific handle exists and you should be fine

```ts
const connectedEdges = getConnectedEdges([node], edges);

const filteredEdges = connectedEdges.filter(edge => edge.sourceHandle === sourceHandleId)
```

**Reply by badiuciprian** [reply]
Thank you for the solution, I will check it out.
